src/extensions/mod.rs

```diff
@@ -25,6 +34,155 @@ impl Default for Extension {
 }
 
 pub trait BasicExtension {
-    fn info(&self) -> Extension;
+    fn info() -> Extension;
     fn render(cx: Scope) -> Element;
 }
+
+#[derive(Clone)]
+pub struct ExtensionManager {
+    pub info: Extension,
+    pub render: fn(Scope) -> Element
+}
+
+impl std::fmt::Display for ExtensionType {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        match self {
+            ExtensionType::ChatbarIcon => write!(f, "ChatbarIcon"),
+            ExtensionType::SidebarWidget => write!(f, "SidebarWidget"), 
+        }
+    }
+}
+
+impl std::fmt::Display for Extension {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        let mut extension_manager_display = String::new();
+
+        extension_manager_display.push_str(self.name.as_str());
+        extension_manager_display.push_str(", \n");
+        extension_manager_display.push_str(self.author.as_str());
+        extension_manager_display.push_str(", \n");
+        extension_manager_display.push_str(self.description.as_str());
+        extension_manager_display.push_str(", \n");
+        extension_manager_display.push_str(self.location.to_string().as_str());
+
+        write!(f, "{}", extension_manager_display)
+    }
+}
+
+
+impl std::fmt::Display for ExtensionManager {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        let mut extension_manager_display = String::new();
+
+        extension_manager_display.push_str(self.info.to_string().as_str());
+
+        write!(f, "{}", extension_manager_display)
+    }
+}
+
+
+pub fn get_extensions() -> HashMap<ExtensionType, Vec<ExtensionManager>>{
+    let mut map_extensions: HashMap<ExtensionType, Vec<ExtensionManager>> = HashMap::new();
+    let mut ext_mng;
+    fs::create_dir_all("extensions").unwrap();
```

Comment:
This probably would need to use a path to this directory as this would create an `extensions` in the current directory where its running at.

Reply:
Oh yes, I didn't notice before, already changed